framework/include/auxkernels/AuxKernel.h

```diff
@@ -66,6 +66,12 @@ class AuxKernel : public MooseObject,
 
   virtual ~AuxKernel();
 
+  /**
+   * Called just before the compute loop starts so an AuxKernel can initialize
+   * any internal datastructures
+   */
+  virtual void initialize() {}
```

Comment:
I have to say that I'm pretty against this. This basically makes the AuxKernel system into a specialized UserObject type system. Have you considered making a user object that can produce this field? Especially, since this is borderline testing. All of the existing RelationshipManager fields are in a UserObject already. If you have several RelationshipManagers, you could store them by name in a map or some other data structure and just have one UserObject to handle them all.  

Since this includes a libMesh update (which has to be changed), we still have time to reconsider this design.